src/Schema/PostgreSQLSchemaManager.php

```diff
@@ -710,71 +698,59 @@ protected function selectForeignKeyColumns(string $databaseName, ?string $tableN
                       FROM pg_catalog.pg_class c, pg_catalog.pg_namespace n
 SQL;
 
-        $conditions = ['n.oid = c.relnamespace'];
-        $params     = [];
-
-        if ($tableName !== null) {
-            $conditions[] = $this->getTableWhereClause($tableName);
-        } else {
-            $conditions[] = "n.nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast')";
-            $conditions[] = 'n.nspname = ANY(current_schemas(false))';
-        }
+        $conditions = array_merge(['n.oid = c.relnamespace'], $this->buildQueryConditions($tableName));
 
         $sql .= ' WHERE ' . implode(' AND ', $conditions) . ") AND r.contype = 'f'";
 
-        return $this->_conn->executeQuery($sql, $params);
+        return $this->_conn->executeQuery($sql);
     }
 
     /**
      * {@inheritDoc}
      */
     protected function fetchTableOptionsByTable(string $databaseName, ?string $tableName = null): array
     {
-        if ($tableName === null) {
-            $tables = $this->listTableNames();
-        } else {
-            $tables = [$tableName];
-        }
+        $sql = <<<'SQL'
+SELECT c.relname,
+       obj_description(c.oid, 'pg_class') AS comment
+FROM pg_class c
+     INNER JOIN pg_namespace n
+         ON n.oid = c.relnamespace
+SQL;
 
-        $tableOptions = [];
-        foreach ($tables as $table) {
-            $sql     = 'SELECT obj_description(?::regclass) AS table_comment;';
-            $comment = $this->_conn->executeQuery($sql, [$table])->fetchOne();
+        $conditions = array_merge(["c.relkind = 'r'"], $this->buildQueryConditions($tableName));
 
-            if ($comment === null) {
-                continue;
-            }
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
 
-            $tableOptions[$table]['comment'] = $comment;
-        }
-
-        return $tableOptions;
+        return $this->_conn->fetchAllAssociativeIndexed($sql);
     }
 
     /**
-     * @param string $table
-     * @param string $classAlias
-     * @param string $namespaceAlias
+     * @param string|null $tableName
+     *
+     * @return list<string>
      */
-    private function getTableWhereClause($table, $classAlias = 'c', $namespaceAlias = 'n'): string
+    private function buildQueryConditions($tableName): array
```

Comment:
```suggestion
    private function buildQueryConditions(?string $tableName): array
```

Reply:
You're right. We don't have strict types enabled here, so even given that the public methods that call this one don't enforce the type of `$tableName`, we can rely on implicit casting here.

But this is non-essential and is hard to reason about since these requirements aren't enforced by the code style or static analysis.

This change will be consistently implemented during the merge up to 4.0.x.